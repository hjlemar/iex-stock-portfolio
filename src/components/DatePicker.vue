<template>
  <v-date-picker width="290" v-model="date" color="indigo"></v-date-picker>
</template>

<script>

import { mapGetters } from 'vuex';
import { CHANGE_DATE as type } from '../store/mutation-types';

const { get } = {
  ...mapGetters({
    get: 'date',
  }),
};

export default {

  computed: {
    date: {
      get,
      set(date) {
        this.$store.commit({ type, date });
      },
    },
  },
};
</script>

<style>

</style>

