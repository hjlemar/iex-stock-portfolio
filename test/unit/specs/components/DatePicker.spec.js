import { shallow, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import DatePicker from '@/components/DatePicker';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('DatePicker.vue', () => {
  let store;
  let getters;

  beforeEach(() => {
    getters = {
      date: jest.fn(),
    };
    store = new Vuex.Store({
      state: {
        date: '2018-02-01',
      },
      getters,
    });
  });

  it('should render component and match snapshot', () => {
    const wrapper = shallow(DatePicker, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
