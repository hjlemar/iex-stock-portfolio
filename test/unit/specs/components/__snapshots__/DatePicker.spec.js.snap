// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker.vue should render component and match snapshot 1`] = `undefined`;
